src/go/rpk/pkg/cli/cmd/redpanda/admin/brokers/brokers.go

```diff
@@ -161,3 +162,39 @@ the cluster leader handles the request.
 		},
 	}
 }
+
+func newDecommissionStatus(fs afero.Fs) *cobra.Command {
+	var completion int
+	return &cobra.Command{
+		Use:   "decommission-status [BROKER ID]",
+		Short: "Show the progress of a node decommissioning",
+		Args:  cobra.ExactArgs(1),
+		Run: func(cmd *cobra.Command, args []string) {
+			broker, _ := strconv.Atoi(args[0])
+			p := config.ParamsFromCommand(cmd)
+			cfg, err := p.Load(fs)
+			out.MaybeDie(err, "unable to load config: %v", err)
+
+			cl, err := admin.NewClient(fs, cfg)
+			out.MaybeDie(err, "unable to initialize admin client: %v", err)
+
+			dbs, err := cl.DecommissionBrokerStatus(cmd.Context(), broker)
+			out.MaybeDie(err, "unable to request brokers: %v", err)
+
+			headers := []string{"Namespace-Topic", "Partition", "Moving-to", "Parttion-size", "Bytes-moved", "Bytes-remaining", "Completion (%)"}
+			f := func(p *admin.DecommissionPartitions) []interface{} {
+				nt := p.Ns + "/" + p.Topic
+				if p.PartitionSize > 0 {
+					completion = p.BytesMoved * 100 / p.PartitionSize
+				}
+				ret := []interface{}{nt, p.Partition, p.MovingTo.NodeID, p.PartitionSize, p.BytesMoved, p.BytesLeftToMove, completion}
```

Comment:
Can this change to:
```
return struct {
    // types
}{
    // fields
}
```
with types & fields each on a dedicated line? And then below, use `PrintStructFields`?

This way, we ensure that any type changes in api_broker cause compile time failures here -- this file can guard itself against API changes in the other file.

(I introduced PrintStructFields after I accidentally caused a type change from a number to a pointer)

Reply:
Wasn't aware of it, thank you!